File: test-examples-app.js
/**
 * App that exercises test-examples-lib functions for type observation.
 */
require('trickle/auto');

const { calculateDiscount, formatAddress, sumArray } = require('./test-examples-lib');

// Call functions with specific values (these will appear as @example)
const d = calculateDiscount(99.99, 15);
console.log('discount:', d.final);

const a = formatAddress('123 Main St', 'Springfield', '62704');
console.log('address:', a.full);

const s = sumArray([10, 20, 30, 40, 50]);
console.log('sum:', s.total);

console.log('Done!');

File: test_examples_app.py
"""App that exercises test_examples_lib functions for type observation."""
import trickle.auto  # noqa: F401

from test_examples_lib import calculate_discount, format_address, sum_array

d = calculate_discount(99.99, 15)
print("discount:", d["final"])

a = format_address("123 Main St", "Springfield", "62704")
print("address:", a["full"])

s = sum_array([10, 20, 30, 40, 50])
print("sum:", s["total"])

print("Done!")
